src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCloneObject = { ...state };
+  const result = [];
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      Object.assign(stateCloneObject, obj.extraData);
+    }
+
+    if (obj.type === 'removeProperties') {
+      for (const key of obj.keysToRemove) {
+        delete stateCloneObject[key];
+      }
+    }
+
+    if (obj.type === 'clear') {
+      for (const key in stateCloneObject) {
+        delete stateCloneObject[key];
+      }
+    }
```

Comment:
Good start but let's rewrite it to switch/case statement since we rely on a single obj.type value